File: src/decode.ts
import * as Amino from '../lib/Amino';

export function decodeByte (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeByte(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeByteSlice (amino: Uint8Array): [Uint8Array, number] {
    const [value, length, error] = Amino.decodeByteSlice(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeInt8 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeInt8(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeInt16 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeInt16(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeInt32 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeInt32(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeInt64 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeInt64(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeUint8 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeUint8(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeUint16 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeUint16(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeUint32 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeUint32(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeUint64 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeUint64(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeVarint (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeVarint(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeUvarint (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeUvarint(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeFloat32 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeFloat32(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeFloat64 (amino: Uint8Array): [number, number] {
    const [value, length, error] = Amino.decodeFloat64(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeBool (amino: Uint8Array): [boolean, number] {
    const [value, length, error] = Amino.decodeBool(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeString (amino: Uint8Array): [string, number] {
    const [value, length, error] = Amino.decodeString(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

export function decodeTime (amino: Uint8Array): [Date, number] {
    const [value, length, error] = Amino.decodeTime(amino);
    if (error) {
        throw error;
    }
    return [value, length];
}

File: go/js/decode.go
package main

import (
	"github.com/cosmos/amino-js/go/src"
	"time"
)

func DecodeByte(bz []byte) (b byte, n int) {
	b, n, err := src.DecodeByte(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeByteSlice(bz []byte) (bz2 []byte, n int) {
	bz2, n, err := src.DecodeByteSlice(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeInt8(bz []byte) (i int8, n int) {
	i, n, err := src.DecodeInt8(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeInt16(bz []byte) (i int16, n int) {
	i, n, err := src.DecodeInt16(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeInt32(bz []byte) (i int32, n int) {
	i, n, err := src.DecodeInt32(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeInt64(bz []byte) (i int64, n int) {
	i, n, err := src.DecodeInt64(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeVarint(bz []byte) (i int64, n int) {
	i, n, err := src.DecodeVarint(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeUint8(bz []byte) (u uint8, n int) {
	u, n, err := src.DecodeUint8(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeUint16(bz []byte) (u uint16, n int) {
	u, n, err := src.DecodeUint16(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeUint32(bz []byte) (u uint32, n int) {
	u, n, err := src.DecodeUint32(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeUint64(bz []byte) (u uint64, n int) {
	u, n, err := src.DecodeUint64(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeUvarint(bz []byte) (u uint64, n int) {
	u, n, err := src.DecodeUvarint(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeFloat32(bz []byte) (f float32, n int) {
	f, n, err := src.DecodeFloat32(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeFloat64(bz []byte) (f float64, n int) {
	f, n, err := src.DecodeFloat64(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeBool(bz []byte) (b bool, n int) {
	b, n, err := src.DecodeBool(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeString(bz []byte) (s string, n int) {
	s, n, err := src.DecodeString(bz)
	if err != nil {
		panic(err)
	}
	return
}

func DecodeTime(bz []byte) (t time.Time, n int) {
	t, n, err := src.DecodeTime(bz)
	if err != nil {
		panic(err)
	}
	return
}
